third_party/proxy/meta_proxier.go

```diff
@@ -160,12 +161,12 @@ func (proxier *metaProxier) OnEndpointsSynced() {
 }
 
 func (proxier *metaProxier) GetServiceByIP(serviceStr string) (ServicePortName, bool) {
-	if utilnet.IsIPv6String(serviceStr) {
+	// Format of serviceStr is fmt.Sprintf("%s:%d/%s", clusterIP, svcPort, protocol), so IPv6 serviceStr has more than 3 colons.
+	if strings.Count(serviceStr, ":") >= 3 {
```

Comment:
Check looks somewhat arbitrary. Can we parse the string with last colon as the separator and use the IP?

Reply:
Thanks, addressed.